autosklearn/ensemble_building/builder.py

```diff
@@ -677,41 +674,84 @@ def candidate_selection(
         for run in discarded:
             self.logger.warning(f"Have no ensemble predictions for {run}")
 
-        # Get all the ones that have a tangible loss
+        # Get all the ones that have all metrics and tangible losses
+        has_metrics = lambda r: all(m.name in r.losses for m in metrics)
```

Comment:
Did this happen to you that not all metrics were present?

Reply:
Nope, but it could technically happen if a metric fails to compute for some reason, either the ensemble builder was cut off or some error occured.